File: Maps/CountMaxPairsWithK-Sum/count.py
from typing import List

class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        totalOps = 0
        mp = {}
        n = len(nums)
        for i in range(n):
            mp[nums[i]] = mp.get(nums[i],0) + 1
        for ele in mp:
            target = k - ele
            if ele == target:
                while mp[ele] > 1:
                    totalOps += 1
                    mp[ele] -= 2
            else:
                if target in mp and ele<target:
                    totalOps += min(mp[ele],mp[target])
        return totalOps
        

File: Maps/CountMaxPairsWithK-Sum/count.cpp
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    int maxOperations(vector<int>& nums, int k) {
        int totalOps = 0;
        unordered_map<int,int>mp;
        int n = nums.size();
        for(int i=0;i<n;i++) mp[nums[i]]++;
        for(auto p: mp){
            int ele = p.first;
            int target = k - ele;
            if(ele == target){
                while(mp[ele] > 1){
                    totalOps += 1;
                    mp[ele] -= 2;
                }
            }
            else
                if(mp.find(target)!=mp.end() && ele<target)
                    totalOps += min(mp[ele],mp[target]);
        }
        return totalOps;
    }
};